src/Azure.Functions.Cli/Helpers/PythonHelpers.cs

```diff
@@ -156,9 +157,16 @@ private static void SetWorkerPathIfNeeded(bool setWorker, string pyExe)
             }
         }
 
+        private static async Task<WorkerLanguageVersionInfo> GetVersion(string pythonExe = "python")
+        {
+            var pythonExeVersionTask = VerifyVersion(pythonExe);
+            var pythonExeVersion = await Utilities.SafeExecution(async () => await pythonExeVersionTask) ?? string.Empty;
+            return new WorkerLanguageVersionInfo(WorkerRuntime.python, pythonExeVersion, pythonExe);
+        }
+
         public static async Task<string> VerifyVersion(string pythonExe = "python")
         {
-            var exe = new Executable(pythonExe, "--version");
+            var exe = new Executable(pythonExe, "-c \"import sys; print(str(sys.version_info[0]) + \'.\' + str(sys.version_info[1]) + \'.\' + str(sys.version_info[2]))\"");
```

Comment:
I would rather stay away from Python code to detect version. Also because syntax has changed over time. This may fail for older versions of Python.